docs/tutorial/flask.rst

```diff
@@ -0,0 +1,535 @@
+will be passed as environment variables to the Flask application. Add the
+following to the end of the ``charmcraft.yaml`` file:
+
+.. code-block:: yaml
+
+    config:
+      options:
+        greeting:
+          description: |
+            The greeting to be returned by the Flask application.
+          default: "Hello, world!"
+          type: string
+
+.. note::
+
+    Configuration options are automatically capitalised and ``-`` are replaced
+    by ``_``. A ``FLASK_`` prefix will also be added which will let Flask
+    identify which environment variables to include when running
+    ``app.config.from_prefixed_env()`` in ``app.py``.
+
+Run ``charmcraft pack`` again. The deployment can now be refreshed to
+make use of the new code:
+
+.. literalinclude:: code/flask/task.yaml
+    :language: bash
+    :start-after: [docs:refresh-deployment]
+    :end-before: [docs:refresh-deployment-end]
+    :dedent: 2
+
+.. note::
+
+    For the refresh command, the ``--constraints`` option is not required if
+    you are not running on an ``amd64`` host as Juju will remember the
+    constraint for the life of the application deployment.
+
+Wait for ``juju status`` to show that the App is ``active`` again. Verify that
+the new configuration has been added using
+``juju config flask-hello-world | grep -A 6 greeting:`` which should show
+the configuration option.
+
+.. note::
+
+    The ``grep`` command extracts a portion of the configuration to make
+    it easier to check whether the configuration option has been added.
+
+Using ``curl http://flask-hello-world --resolve flask-hello-world:80:127.0.0.1``
+shows that the response is still ``Hello, world!`` as expected.
+The greeting can be changed using Juju:
+
+.. literalinclude:: code/flask/task.yaml
+    :language: bash
+    :start-after: [docs:change-config]
+    :end-before: [docs:change-config-end]
+    :dedent: 2
+
+``curl http://flask-hello-world --resolve flask-hello-world:80:127.0.0.1``
+now returns the updated ``Hi!`` greeting.
+
+.. note::
+
+    It might take a short time for the configuration to take effect.
+
+Integrate with a database
+=========================
+
+Now let's keep track of how many visitors your application has received.
+This will require integration with a database to keep the visitor count.
+This will require a few changes:
+
+* We will need to create a database migration that creates the ``visitors`` table
+* We will need to keep track how many times the root endpoint has been called
+in the database
+* We will need to add a new endpoint to retrieve the number of visitors from the
+database
+
+The charm created by the ``flask-framework`` extension will execute the
+``migrate.py`` script if it exists. This script should ensure that the
+database is initialised and ready to be used by the application. We will
+create a ``migrate.py`` file containing this logic.
+
+Go back out to the tutorial root directory using ``cd ..``. Open the ``migrate.py``
```

Comment:
It looks a bit weird to have `cd ..` followed by a period which makes it seem like we are telling people to do `cd ...`